homeassistant/components/switcher_kis/climate.py

```diff
@@ -0,0 +1,240 @@
+            connections={
+                (device_registry.CONNECTION_NETWORK_MAC, coordinator.mac_address)
+            }
+        )
+
+        self._attr_min_temp = remote.min_temperature
+        self._attr_max_temp = remote.max_temperature
+        self._attr_target_temperature_step = 1
+        self._attr_temperature_unit = TEMP_CELSIUS
+
+        self._attr_supported_features = 0
+        self._attr_hvac_modes = [HVACMode.OFF]
+        for mode in remote.modes_features:
+            self._attr_hvac_modes.append(DEVICE_MODE_TO_HA[mode])
+            features = remote.modes_features[mode]
+
+            if features["temperature_control"]:
+                self._attr_supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE
+
+            if features["fan_levels"]:
+                self._attr_supported_features |= ClimateEntityFeature.FAN_MODE
+
+            if features["swing"] and not remote.separated_swing_command:
+                self._attr_supported_features |= ClimateEntityFeature.SWING_MODE
+
+        self._update_data(True)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_data()
+        self.async_write_ha_state()
+
+    def _update_data(self, force_update: bool = False) -> None:
+        """Update data from device."""
+        data = self.coordinator.data
+        features = self._remote.modes_features[data.mode]
+
+        if data.target_temperature == 0 and not force_update:
+            return
+
+        self._attr_current_temperature = cast(float, data.temperature)
+        self._attr_target_temperature = float(data.target_temperature)
+
+        self._attr_hvac_mode = HVACMode.OFF
+        if data.device_state == DeviceState.ON:
+            self._attr_hvac_mode = DEVICE_MODE_TO_HA[data.mode]
+
+        self._attr_fan_mode = None
+        self._attr_fan_modes = []
+        if features["fan_levels"]:
+            self._attr_fan_modes = [DEVICE_FAN_TO_HA[x] for x in features["fan_levels"]]
+            self._attr_fan_mode = DEVICE_FAN_TO_HA[data.fan_level]
+
+        self._attr_swing_mode = None
+        self._attr_swing_modes = []
+        if features["swing"]:
+            self._attr_swing_mode = SWING_OFF
+            self._attr_swing_modes = [SWING_VERTICAL, SWING_OFF]
+            if data.swing == ThermostatSwing.ON:
+                self._attr_swing_mode = SWING_VERTICAL
+
+    async def _async_control_breeze_device(
+        self,
+        state: DeviceState | None = None,
+        mode: ThermostatMode | None = None,
+        target_temp: int | None = None,
+        fan_mode: ThermostatFanLevel | None = None,
+        swing_mode: ThermostatSwing | None = None,
+    ) -> None:
+        """Call Switcher Control Breeze API."""
+        response: SwitcherBaseResponse = None
+        error = None
+        current_state: SwitcherThermostat = self.coordinator.data
+
+        try:
+            command = self._remote.get_command(
```

Comment:
Could this technically throw an exception? If not, put it outside the try/except so it's really clear where the exception could arise.

Reply:
Yes, this can throw `RuntimeError` if the command does not exists, in theory it can happen if the integration is trying to request a command which is not supported, most likely a bug in the integration or upstream report incorrect device capabilities